### DISCUSSION TITLE
force_update

### DISCUSSION THREAD

**Discussion by Kirono** [Hyundai-Kia-Connect/kia_uvo#257] [Q&A]
It is stated that this function should not be overused, how often would be considered an overuse? More than once every 4 hours as default or once a minute. 

What are the consequences of overusing this feature? Is it a matter of software limitation causing bugs or getting banned from servers or mabey draining 12v battery?

**Comment by cdnninja** [comment]
Yes it can drain the 12v battery as it does poll the car for data.   You may also max out API calls as most regions have a daily limit on how much you can use the car.  More on this here: https://github.com/Hacksore/bluelinky/wiki/API-Rate-Limits

**Reply by alexwhittemore** [reply]
To be clear, these rate limits (and the concept of a forced update draining battery) only apply to forced updates, right? Whereas a normal update is merely getting the last reported value from the web service, which may have been pushed from the car or caused by the user force-updating from the Kia/Hyundai app itself?

In other words, there’s LIKELY no disadvantage (apart from maybe unknown rate limits) to setting the update rate very frequent and the forced rate very slow?

also- forced updates seem to drain battery because the car doesn’t enter ultra-low-power mode for some extended period following a forced update. In other words, forcing an update doesn’t drain the car - the car waiting in a high-power “waiting for another poll” state because it recently got polled drains power. Does anyone know anything about the timing characteristics of these power modes, like how long it takes a deep-sleeping car to respond to a forced update or how long after one it goes into deep sleep?

**Reply by rage2021** [reply]
any answer on this? Im keen to know as well. What if I forced every 45 mins?

On top of this sometimes I go back to home assistant and its just not accurate unless I do a force update. not sure what happens with the forced refresh (Canada)

**Comment by arjanhs** [comment]
To overcome this there coult be build in an control mechanisme which checks the 12v battery before doing an actual poll to the car. When the power in the 12v battery is to low, no poll should be made to the car but only cached information should be used.

**Comment by Zaphod360** [comment]
Is there a way to do a manual forced update? I'm happy with the default refresh rate but occasionally it would be nice to pull the live info.

**Reply by Ricks88** [reply]
I would like this also!

**Reply by cdnninja** [reply]
This is a service exposed you can call.   Can be accessed via automation or developer menu.   Don't over use it though as api limits exist with Hyundai and Kia.

**Reply by Ricks88** [reply]
Can i create a Automation or 'button' the manualy refresh?
Any example maybe, i never did this yet :P

**Reply by rage2021** [reply]
also wondering this.

**Comment by Ricks88** [comment]
Can someone give me an example of a force update 'automation' or something else please?

**Reply by richy58729** [reply]
Lovelace card with a button:

``` yaml
show_name: true
show_icon: true
type: button
tap_action:
  action: call-service
  service: kia_uvo.force_update
  data: {}
  target: {}
name: Force update
icon: mdi:car-connected
show_state: false
hold_action:
  action: none
```

Is that helpful?

**Reply by Ricks88** [reply]
> Lovelace card with a button:
> 
> ```yaml
> show_name: true
> show_icon: true
> type: button
> tap_action:
>   action: call-service
>   service: kia_uvo.force_update
>   data: {}
>   target: {}
> name: Force update
> icon: mdi:car-connected
> show_state: false
> hold_action:
>   action: none
> ```
> 
> Is that helpful?

Yes, Thank you !

Thought i needed to create a full Automation, did not know i could do it in a button :)

**Comment by Vagabond80** [comment]
I recently managed to drain the car battery of my Sportage PHEV. I guess it was due to me working on HASS a lot and restarting it very often. Am I wrong or does this trigger a force_update? I did not move the car for a couple of days and one morning it was empty. I set the force stop and start to the same time so that no force_updates are being done and rely completely on the cloud updates, since this is enough for what I do with the car data. I can only emphasize to be careful with excessive force_update.

**Reply by cdnninja** [reply]
Just checked.  On restart atleast mine does a cached call.  Now some regions will do a location lookup on start (Canada) as no cached data exists for location in Canada.    After that Canada won't lookup location until cached data shows an odometer change.  

I believe most regions don't do that though.  Logs can confirm this.

**Comment by GarryG** [comment]
Having had an issue due to too many updates draining my 12V battery, is there a way to disable forced updates completely? I set the "no forced updates" start and end times to 0 and 23 - would that prevent forced updates?

**Reply by cdnninja** [reply]
Which region? I would encourage you to search logs to see if any force updates occur.   Limited situations can cause it like in Canada to get location if odometer increased.
